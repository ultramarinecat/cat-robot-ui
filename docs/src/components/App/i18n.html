<!DOCTYPE html>

<html>
<head>
  <title>i18n.js</title>
  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
  <meta name="viewport" content="width=device-width, target-densitydpi=160dpi, initial-scale=1.0; maximum-scale=1.0; user-scalable=0;">
  <link rel="stylesheet" media="all" href="../../../docco.css" />
</head>
<body>
  <div id="container">
    <div id="background"></div>
    
      <ul id="jump_to">
        <li>
          <a class="large" href="javascript:void(0);">Jump To &hellip;</a>
          <a class="small" href="javascript:void(0);">+</a>
          <div id="jump_wrapper">
          <div id="jump_page_wrapper">
            <div id="jump_page">
              
                
                <a class="source" href="App.html">
                  src/components/App/App.js
                </a>
              
                
                <a class="source" href="ErrorBoundary.html">
                  src/components/App/ErrorBoundary.js
                </a>
              
                
                <a class="source" href="i18n.html">
                  src/components/App/i18n.js
                </a>
              
                
                <a class="source" href="../Auth/Auth.html">
                  src/components/Auth/Auth.js
                </a>
              
                
                <a class="source" href="../Auth/AuthActions.html">
                  src/components/Auth/AuthActions.js
                </a>
              
                
                <a class="source" href="../Auth/AuthReducer.html">
                  src/components/Auth/AuthReducer.js
                </a>
              
                
                <a class="source" href="../Button/Button.html">
                  src/components/Button/Button.js
                </a>
              
                
                <a class="source" href="../Cat/Cat.html">
                  src/components/Cat/Cat.js
                </a>
              
                
                <a class="source" href="../Cat/CatActions.html">
                  src/components/Cat/CatActions.js
                </a>
              
                
                <a class="source" href="../Cat/CatReducer.html">
                  src/components/Cat/CatReducer.js
                </a>
              
                
                <a class="source" href="../Cat/fulltilt.html">
                  src/components/Cat/fulltilt.js
                </a>
              
                
                <a class="source" href="../Connection/Connection.html">
                  src/components/Connection/Connection.js
                </a>
              
                
                <a class="source" href="../Connection/ConnectionActions.html">
                  src/components/Connection/ConnectionActions.js
                </a>
              
                
                <a class="source" href="../Connection/ConnectionReducer.html">
                  src/components/Connection/ConnectionReducer.js
                </a>
              
                
                <a class="source" href="../Connection/messages.html">
                  src/components/Connection/messages.js
                </a>
              
                
                <a class="source" href="../Intro/Intro.html">
                  src/components/Intro/Intro.js
                </a>
              
                
                <a class="source" href="../Intro/IntroActions.html">
                  src/components/Intro/IntroActions.js
                </a>
              
                
                <a class="source" href="../Particles/Particles.html">
                  src/components/Particles/Particles.js
                </a>
              
                
                <a class="source" href="../Pulse/Pulse.html">
                  src/components/Pulse/Pulse.js
                </a>
              
                
                <a class="source" href="../Router/Router.html">
                  src/components/Router/Router.js
                </a>
              
                
                <a class="source" href="../Sound/CatSound.html">
                  src/components/Sound/CatSound.js
                </a>
              
                
                <a class="source" href="../Sound/Mute.html">
                  src/components/Sound/Mute.js
                </a>
              
                
                <a class="source" href="../Sound/Sound.html">
                  src/components/Sound/Sound.js
                </a>
              
                
                <a class="source" href="../Sound/TurningSound.html">
                  src/components/Sound/TurningSound.js
                </a>
              
                
                <a class="source" href="../Spinner/Spinner.html">
                  src/components/Spinner/Spinner.js
                </a>
              
                
                <a class="source" href="../Typing/Typing.html">
                  src/components/Typing/Typing.js
                </a>
              
                
                <a class="source" href="../../configureStore.html">
                  src/configureStore.js
                </a>
              
                
                <a class="source" href="../../index.html">
                  src/index.js
                </a>
              
                
                <a class="source" href="../../serviceWorker.html">
                  src/serviceWorker.js
                </a>
              
                
                <a class="source" href="../../setupTests.html">
                  src/setupTests.js
                </a>
              
            </div>
          </div>
        </li>
      </ul>
    
    <ul class="sections">
        
          <li id="title">
              <div class="annotation">
                  <h1>i18n.js</h1>
              </div>
          </li>
        
        
        
        <li id="section-1">
            <div class="annotation">
              
              <div class="pilwrap ">
                <a class="pilcrow" href="#section-1">&#182;</a>
              </div>
              <p>i18n
Configures i18n, adds en and es translations</p>

            </div>
            
            <div class="content"><div class='highlight'><pre>
<span class="hljs-keyword">import</span> i18n <span class="hljs-keyword">from</span> <span class="hljs-string">'i18next'</span>;
<span class="hljs-keyword">import</span> LanguageDetector <span class="hljs-keyword">from</span> <span class="hljs-string">'i18next-browser-languagedetector'</span>;
<span class="hljs-keyword">import</span> { initReactI18next } <span class="hljs-keyword">from</span> <span class="hljs-string">'react-i18next'</span>;

<span class="hljs-keyword">import</span> en <span class="hljs-keyword">from</span> <span class="hljs-string">'./locales/en/common.json'</span>;
<span class="hljs-keyword">import</span> es <span class="hljs-keyword">from</span> <span class="hljs-string">'./locales/es/common.json'</span>;

i18n
  .use(LanguageDetector) <span class="hljs-comment">// detect language</span>
  .use(initReactI18next)
  .init({
    <span class="hljs-attr">resources</span>: { <span class="hljs-attr">cat</span>: <span class="hljs-string">'meow'</span> }, <span class="hljs-comment">// (needs to initialize with a resource)</span>
    fallbackLng: <span class="hljs-string">'en'</span>, <span class="hljs-comment">// default lang</span>
    keySeparator: <span class="hljs-literal">false</span>,
    <span class="hljs-attr">interpolation</span>: {
      <span class="hljs-attr">escapeValue</span>: <span class="hljs-literal">false</span>
    }
  });</pre></div></div>
            
        </li>
        
        
        <li id="section-2">
            <div class="annotation">
              
              <div class="pilwrap ">
                <a class="pilcrow" href="#section-2">&#182;</a>
              </div>
              <p>add english and spanish translations</p>

            </div>
            
            <div class="content"><div class='highlight'><pre>i18n.addResources(<span class="hljs-string">'en'</span>, <span class="hljs-string">'translation'</span>, en);
i18n.addResources(<span class="hljs-string">'es'</span>, <span class="hljs-string">'translation'</span>, es);

<span class="hljs-keyword">export</span> <span class="hljs-keyword">default</span> i18n;</pre></div></div>
            
        </li>
        
    </ul>
  </div>
</body>
</html>
