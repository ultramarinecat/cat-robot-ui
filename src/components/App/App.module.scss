@charset 'utf-8';

////
//// App styles
////

@import 'base.scss';

$background-color: $indigo-500;
$background-image: 'url(./cat.jpg)';

html {
  // height, width 100%;
  width: $full-parent-width;
  height: $full-parent-height;
  overflow: hidden;

  // smooth fonts on chrome, mobile safari
  -webkit-font-smoothing: antialiased;
}

body {
  // full-page background image
  background: #{$background-image} $background-color no-repeat center center fixed;
  background-size: cover;

  width: $full-parent-width;
  height: $full-parent-height;
  position: relative;
  overflow: hidden;
}

main {
  // center content vertically and horizontally
  position: absolute;
  top: $half-parent-height;
  left: $half-parent-width;
  transform: translate(-$half-parent-width, -$half-parent-height);

  width: $full-parent-width;
  @extend %center-block;
  @extend %center-inline;
}

.logo {
  font-family: $logo-font;
  font-style: italic;

  font-size: rem(54.4);
  font-weight: 700;

  @media #{$tablet-plus} {
    font-size: rem(60.8);
  }

  padding-top: 10vh;
  text-shadow: rem(1) rem(6) rem(3) rgba(#000, 0.1);
  // text-shadow: rem(1.6) rem(8) rem(2.6) rgba(#000, 0.12);
}

// selection color
::-moz-selection {
  background: $selection-background-color;
  text-shadow: none;
}

::selection {
  background: $selection-background-color;
  text-shadow: none;
}
