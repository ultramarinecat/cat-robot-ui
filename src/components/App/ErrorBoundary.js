//
// Main error boundary, displays fallback ui indicating that an error has occurred
//

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { withTranslation } from 'react-i18next';

import styles from './ErrorBoundary.module.scss';

export class ErrorBoundary extends Component {
  state = { hasError: false };

  static propTypes = {
    t: PropTypes.func.isRequired
  };

  static getDerivedStateFromError() {
    return { hasError: true };
  }

  render() {
    // if error, display fallback ui
    if (this.state.hasError) {
      const { t } = this.props;

      return (
        <div className={styles.main} data-testid="error-fallback-ui">
          {t('error')}
        </div>
      );
    }

    return this.props.children;
  }
}

export default withTranslation()(ErrorBoundary);
