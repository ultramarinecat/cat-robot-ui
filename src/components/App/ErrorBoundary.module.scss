@charset 'utf-8';

////
//// Error boundary styles
////

@import 'base.scss';

.main {
  position: absolute;
  top: $half-parent-height;
  left: $half-parent-width;
  transform: translate(-$half-parent-width, -$half-parent-height);
  width: $full-parent-width;
  @extend %center-inline;

  padding: rem(15);
  font-size: rem(18);

  @media #{$tablet-plus} {
    font-size: rem(20);
  }
}
