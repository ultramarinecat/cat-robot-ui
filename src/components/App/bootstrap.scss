////
//// Bootstrap
////

@import 'variables.scss';
@import 'bootstrapOverrides.scss';
@import 'bootstrap/scss/bootstrap.scss';
