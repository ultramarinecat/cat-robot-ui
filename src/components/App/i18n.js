//
// i18n
// Configures i18n, adds en and es translations
//

import i18n from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import { initReactI18next } from 'react-i18next';

import en from './locales/en/common.json';
import es from './locales/es/common.json';

i18n
  .use(LanguageDetector) // detect language
  .use(initReactI18next)
  .init({
    resources: { cat: 'meow' }, // (needs to initialize with a resource)
    fallbackLng: 'en', // default lang
    keySeparator: false,
    interpolation: {
      escapeValue: false
    }
  });

// add english and spanish translations
i18n.addResources('en', 'translation', en);
i18n.addResources('es', 'translation', es);

export default i18n;
