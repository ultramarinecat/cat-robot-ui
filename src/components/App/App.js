//
// App
//

import React, { memo } from 'react';
import PropTypes from 'prop-types';

import { withTranslation } from 'react-i18next';
import log from 'loglevel';

import Particles from '../Particles/Particles.js';
import Router from '../Router/Router.js';
import './i18n.js';

import './bootstrap.scss';
import styles from './App.module.scss';

log.setDefaultLevel(process.env.REACT_APP_LOG_LEVEL);

export function App({ t }) {
  return (
    <div className={styles.main}>
      <Particles />
      <nav>
        <h1 className={`${styles.logo} text-center`}>{t('app_name')}</h1>
      </nav>
      <main>
        <div className="container-fluid">
          <Router />
        </div>
      </main>
    </div>
  );
}

App.propTypes = {
  t: PropTypes.func.isRequired
};

export default memo(withTranslation()(App));
