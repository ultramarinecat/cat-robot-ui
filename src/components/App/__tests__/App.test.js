import React from 'react';
import { render } from 'react-testing-library';

import { App } from '../App.js';

jest.mock('loglevel');
jest.mock('../i18n.js');
jest.mock('../../Particles/Particles');

jest.mock('../../Router/Router', () => () => <div data-testid="router" />);

describe('App', () => {
  it('should render the app router', () => {
    const { container, getByTestId } = render(<App t={key => key} />);

    expect(getByTestId('router')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
