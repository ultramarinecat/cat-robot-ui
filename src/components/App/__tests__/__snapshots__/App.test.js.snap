// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the app router 1`] = `
<div
  class="main"
>
  <nav>
    <h1
      class="logo text-center"
    >
      app_name
    </h1>
  </nav>
  <main>
    <div
      class="container-fluid"
    >
      <div
        data-testid="router"
      />
    </div>
  </main>
</div>
`;
